src/tools/run-make-support/src/external_deps/cygpath.rs

```diff
@@ -22,14 +20,17 @@ use crate::util::handle_failed_output;
 #[track_caller]
 #[must_use]
 pub fn get_windows_path<P: AsRef<Path>>(path: P) -> String {
-    let caller = panic::Location::caller();
-    let mut cygpath = Command::new("cygpath");
-    cygpath.arg("-w");
-    cygpath.arg(path.as_ref());
-    let output = cygpath.run();
-    if !output.status().success() {
-        handle_failed_output(&cygpath, output, caller.line());
+    // If the path looks unixy then use cygpath otherwise return it unchanged.
+    // If cygpath fails then fallback to just using the path as given.
+    if path.as_ref().starts_with("/") {
+        let mut cygpath = Command::new("cygpath");
+        cygpath.arg("-w");
+        cygpath.arg(path.as_ref());
+        let output = cygpath.run();
+        if output.status().success() {
+            // cygpath -w can attach a newline
+            return output.stdout_utf8().trim().to_string();
+        }
```

Comment:
Suggestion: I'm pretty sure `Command::run()` actually asserts exit status is success, so I think this can just be

```suggestion
        let output = cygpath.run().stdout_utf8().trim().to_string();
```

Reply:
EDIT: nvm, that's a special case